bottlerocket/agents/src/bin/ec2-resource-agent/ec2_provider.rs

```diff
@@ -245,7 +245,11 @@ where
                 .min_count(instance_count)
                 .max_count(instance_count)
                 .subnet_id(subnet_id)
-                .set_security_group_ids(Some(spec.configuration.security_groups.clone()))
+                .set_security_group_ids(if spec.configuration.security_groups.is_empty() {
+                    None
```

Comment:
Is this a good idea? I'd like to understand more about this. From reading the code, this sounds like when we initiate the `ec2_client`, we are ok attaching no security groups to it. My instinct is that we should instead fail if there are no security groups since we'd likely always want to have a sane default.

Reply:
For EKS we expect there to be security groups, for ECS we do not expect security groups. `None` is the default.

Reply:
Makes sense!!! 👏